components/ui/ProgressFreemiumComponent.tsx

```diff
@@ -0,0 +1,39 @@
+import { Indicator, Root } from "@radix-ui/react-progress";
+import React, { useEffect, useState } from "react";
+
+export interface ProgressFreemiumComponentProps {
+  elapsedDays: number;
+  totalTrialDays: number;
+}
+const ProgressBarComponent: React.FC<ProgressFreemiumComponentProps> = ({
+  elapsedDays,
+  totalTrialDays,
+}) => {
+  const [progress, setProgress] = useState(0);
+  useEffect(() => {
```

Comment:
Why do you need the state and effect hook? If you're not changing state you can just directly use the prop that you're passing in

Reply:
great point.  I am now relying on the prop elapsedDays